sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -390,6 +394,11 @@ object ParquetReadSupport extends Logging {
       .named(parquetRecord.getName)
   }
 
+  private def clipVariantSchema(parquetType: GroupType, variantStruct: StructType): GroupType = {
+    // TODO(SHREDDING): clip `parquetType` to retain the necessary columns.
```

Comment:
So this requires the new parquet version to support column pruning for variant?

Reply:
I don't think it requires a new Parquet version - it should be possible to clip it in `sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala` in the same way that unused struct fields are clipped. The logic of deciding which fields can be clipped is more complicated, though.

Reply:
It doesn't. in this function, we will have custom logic to clip `parquetType` to retain the the necessary columns for reading `variantStruct`. But this part will be in a future PR to avoid making the single PR too big.